torchx/lib/torchx/backend.ex

```diff
@@ -150,6 +150,44 @@ defmodule Torchx.Backend do
     NIF.transpose(to_ref(t), 0, 1) |> from_ref(out)
   end
 
+  @impl true
+  def slice(%T{shape: shape} = out, %T{} = t, start_indices, lengths, strides) do
+    ref = to_ref(t)
+
+    for dim <- 0..(length(start_indices) - 1), reduce: ref do
+      ref -> NIF.narrow(ref, dim, Enum.at(start_indices, dim), Enum.at(lengths, dim)) |> unwrap!()
+    end
+    |> NIF.as_strided(shape, steps_to_strides(lengths, strides), 0)
```

Comment:
Similarly, there is nothing to stride if `Enum.all?(strides, & &1 == 1)`, so it may be worth adding such check.